codec-http2/src/main/java/io/netty/handler/codec/http2/StreamBufferingEncoder.java

```diff
@@ -198,8 +214,20 @@ public void remoteSettings(Http2Settings settings) throws Http2Exception {
 
     @Override
     public void close() {
-        super.close();
-        cancelPendingStreams();
+        try {
+            if (!closed) {
+                closed = true;
+
+                // Fail all buffered streams.
+                ChannelClosedException e = new ChannelClosedException();
+                for(PendingStream pendingStream : pendingStreams.values()) {
```

Comment:
why not use `pendingStreams.pollFirstEntry`as before ? This will safe us the iterator :)